src/cmake/modules/GUDHI_third_party_libraries.cmake

```diff
@@ -52,12 +52,36 @@ option(WITH_GUDHI_USE_TBB "Build with Intel TBB parallelization" ON)
 
 # Find TBB package for parallel sort - not mandatory, just optional.
 if(WITH_GUDHI_USE_TBB)
-  set(TBB_FIND_QUIETLY ON)
-  find_package(TBB)
-  if (TBB_FOUND)
-    include(${TBB_USE_FILE})
-    message("TBB found in ${TBB_LIBRARY_DIRS}")
-    add_definitions(-DGUDHI_USE_TBB)
+  find_package(TBB QUIET)
+  # cf. https://github.com/oneapi-src/oneTBB/blob/master/cmake/README.md
+  # if (TARGET TBB::tbb) # does not work...
```

Comment:
Can you describe what was going wrong? The error message seemed to be about get_target_property(TBB::tbbmalloc), but if it was the `if` test that returned the wrong answer, I would have expected an error on the previous line. I am confused how changing just the `if` can have fixed things.

Reply:
As it is confusing, I tried to explain better what I meant on [55fef38](https://github.com/GUDHI/gudhi-devel/pull/836/commits/55fef38c2fb757edf02ec113fd35b312e4fa03e6)

Reply:
But in what way did it not work? Did it answer yes when TBB was not found, or no when TBB was actually found? Was it bad on all platforms?

Reply:
IIUC, looking at the "pip build linux" CI, the version with if(TBB_FOUND) succeeds because it decides that it has **not** found TBB.

Reply:
In the case `pip build linux`, TBB is not installed (cf. [Dockerfile](https://github.com/GUDHI/gudhi-deploy/blob/main/Dockerfile_for_pip) - I also checked in `/usr/include` and `/usr/local/include` after a fresh docker pull).
Right after `find_package(TBB)`, `if(TARGET TBB::tbb)` returns true, while `if (TBB_FOUND)` returns false. On this image, cmake is in version 3.12 (July 2018). This may be the root cause. Cmake minimal version is still 3.8 for GUDHI. I would keep the `if (TBB_FOUND)` version. Do you agree ?

Reply:
Ah, `find_package(TBB)` ends up using the FindTBB.cmake from CGAL, and that file seems to have a strange behavior. `find_package(TBB CONFIG)` would hopefully be immune to that. I don't remember if we ever use the TBB functionalities in CGAL, and if this might cause trouble. I am not against sticking to TBB_FOUND, but I find it worth understanding.